### PR TITLE
update to use waf policy

### PR DESCRIPTION
**Pull request by vhorne**
# Best Practice Checklist

Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment.
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: ag-docs-wafv2/azuredeploy.json [hunk 1]
```diff
@@ -1,291 +1,401 @@
...
+                            "destinationAddressPrefix": "*",
+                            "access": "Allow",
+                            "priority": 300,
+                            "direction": "Inbound"
+                        }
                     }
-                }
+                ]
+            },
+            "copy": {
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json:68]: move the copy property above the properties property in all the resources

File: ag-docs-wafv2/azuredeploy.json [hunk 2]
```diff
@@ -1,291 +1,401 @@
...
+            ],
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('vmSize')]"
+                },
+                "storageProfile": {
+                    "imageReference": {
+                        "publisher": "MicrosoftWindowsServer",
+                        "offer": "WindowsServer",
+                        "sku": "2016-Datacenter",
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json:140]: any reason we can't use 2019?

File: ag-docs-wafv2/azuredeploy.json [hunk 3]
```diff
@@ -1,291 +1,401 @@
...
+            "name": "[variables('applicationGateways_myAppGateway_name')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworks_myVNet_name'))]",
+                "[resourceId('Microsoft.Network/publicIPAddresses', concat(variables('publicIPAddress'), '0'))]",
+                "[resourceId('Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies', variables('AGWafPol01'))]"
+            ],
+            "properties": {
+                "sku": {
+                    "name": "WAF_v2",
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json:210]: why did we move these to literals (from parameters)?

File: ag-docs-wafv2/azuredeploy.json (hunk 1 above)
Comment (vhorne) [reply ag-docs-wafv2/azuredeploy.json:68]: done

File: ag-docs-wafv2/azuredeploy.json (hunk 2 above)
Comment (vhorne) [reply ag-docs-wafv2/azuredeploy.json:140]: this needs to be consistent with my other existing quickstart articles, which use 2016

File: ag-docs-wafv2/azuredeploy.json (hunk 3 above)
Comment (vhorne) [reply ag-docs-wafv2/azuredeploy.json:210]: because now I have a policy object, which depends on a waf v2 sku, so I don't want this to be a parameter.

**Comment by vhorne** [comment]
@bmoore-msft 
Changes are in. Need a merge today for pending documentation.

**Pull request merged**